R/plot_xctr_company_level.R

```diff
@@ -0,0 +1,48 @@
+#' Plot xctr data on a company level
+#'
+#' @param data A data frame like [xctr_toy_data].
+#' @param company_name A string. Name of one company in the data set.
+#'
+#' @return A [ggplot] object.
+#' @export
+#'
+#' @examples
+#' plot_xctr_company_level(xctr_toy_data, "company_a")
+plot_xctr_company_level <- function(data, company_name) {
+
+  data <- data |>
+    na.omit() |>
+    filter(company_name == {{ company_name }})
+
+  score_colors <- c("low" = "#007F00", "medium" = "#FFC300", "high" = "#FF5733")
+  risk_category_order <- c("low", "medium", "high")
+
+  share_var <- names(data)[grep("_share", names(data))]
```

Comment:
Maybe use the tidyselect approach?

```r
share_var <- names(select(data, matches("_share"))
```

Remember to check that any column you use in the code exists in the `data`. You may use `hasName()` or something more complex as `check_crucial_names()`.